### PR TITLE
TDD PR

### PR DESCRIPTION
**Pull request by adamteddychang**
In this PR we : 
- Wrote tests with RSpec
- wrote methods to accommodate the tests 
- This project is done in collaboration with @xtrahuman

### CODE REVIEW

File: solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  context 'it should solve problems correctly with methods' do
+    solver = Solver.new
+    string1 = 'hello'
+    string2 = 'adam'
+    string3 = 'tochukwu'
+
+    it 'get factorial of number' do
+      expect(solver.factorial(5)).to eq 120
+      expect(solver.factorial(1)).to eq 1
+      expect { solver.factorial(-1) }.to raise_error('Number is negative')
```
Comment (TedLivist) [inline solver_spec.rb]: - [x] Your tests for the factorial method should contain a special case scenario for when **`0`** is the input - the result should be **`1`**. Kindly add it.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @adamteddychang and @xtrahuman,

Good job so far!
✔️ Good documentation
✔️ Tests were written before actually writing the code.
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
#### _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: solver_spec.rb (hunk 1 above)
Comment (adamteddychang) [reply solver_spec.rb]: I have added the test case, thank you for the review.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @adamteddychang, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**